src/abuf.c

```diff
@@ -73,7 +73,29 @@ abuf_printf(abuf_t *abuf, const char *fmt, ...)
 		if (r < 0)
 			return r;
 	}
-		
+
 	va_end(ap);
 	return 0;
 }
+
+/**
+ * Read string to ending of abstract buffer
+ * @returns 0 if OK, or negative value on error.
+*/
+int
+abuf_append(abuf_t *abuf, const char *src)
+{
+	char *ptr = abuf->p + abuf->pos;
+	size_t size = abuf->len - (ptr - abuf->p);
+	size_t nread = strlcpy(ptr, src, size);
+	if (nread > size--) {
+		int ret = abuf_setup(abuf,
+			abuf->len + ABUF_CHUNK * (nread / ABUF_CHUNK + 1));
```

Comment:
This growth by chunks should be abstracted away somehow. Could be done by `abuf_setup`, but I fear trying to be smarter than the allocator may be counterproductive... in any case this call shold be `abuf_setup(abuf, nread);`